flang/lib/Lower/FIRBuilder.cpp

```diff
@@ -421,6 +425,22 @@ struct CharacterOpsBuilderImpl {
     }
   }
 
+  Char createConcatenate(Char lhs, Char rhs) {
+    mlir::Value len = builder.createHere<mlir::AddIOp>(lhs.len, rhs.len);
+    auto temp = createTemp(rhs.getCharacterType(), len);
+    createCopy(temp, lhs, lhs.len);
+    auto one = builder.createIntegerConstant(len.getType(), 1);
+    auto upperBound = builder.createHere<mlir::SubIOp>(len, one);
+    builder.createLoop(
+        lhs.len, upperBound, one,
+        [&](Fortran::lower::FirOpBuilder &handler, mlir::Value index) {
+          CharacterOpsBuilderImpl charHandler{handler};
+          auto charVal = charHandler.createLoadCharAt(rhs, index);
```

Comment:
A character at rhs[index] should go into temp[lhs.len + index] right?
But here rhs[index] is getting copied into temp[index], how does it work?

Reply:
I can see that `index` value iterates over `[lhs.len, lhs,len + rhs.len - 1]`.  In which case 
`rhs[index-lhs.len]` should get stored into `lhs[index]`.

Reply:
The code absolutely incorrect, thanks, will push end-to-end test to avoid other issues. Thanks !

Reply:
Fixed, and test added.